src/main/java/com/thebois/persistence/AbstractPersistence.java

```diff
@@ -0,0 +1,33 @@
+package com.thebois.persistence;
+
+import java.io.File;
+
+/**
+ * Basic foundation for persistence, common functionality for all persistence classes.
+ */
+abstract class AbstractPersistence {
+
+    private final String directory =
+        System.getProperty("user.home") + "/Documents/Pawntastic/saves/";
+
+    /**
+     * All subclasses need to use this in the constructor to ensure the directory is there before
+     * any commands/operations can be carried out.
+     */
```

Comment:
Just call it in the constructor here, and all subtypes will have to call the `super` constructor. Then this method can be private as well.

Reply:
Well, first you'd need to make the super-constructor protected.

But actually, now that I think about it, doesn't the super-constructor get called automatically if it requires no parameters?

Reply:
Yup, "**If a constructor does not explicitly invoke a superclass constructor, the Java compiler automatically inserts a call to the no-argument constructor of the superclass**"